sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py

```diff
@@ -375,6 +379,7 @@ def delete_container(
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
     def acquire_lease(
```

Comment:
This function calls into the Lease.aquire function which you have decorated separately - does that matter?

(This comment applies to a few different places. I wont comment on every one unless you let me know if it's an issue, in which case I can let you know all the places it applies)

Reply:
so the idea is that there should only be two spans. The span for the entry point function and the span for the network. Is the customer ever expected to call aquire_lease? If not I think the best thing to do would be to just not decorate any of them

Reply:
Yes the customer is expected to call this - it's just that there's two paths to access the same function:
a) Customer directly calls Lease.aquire_lease
b) Customer calls Blob or ContainerClient.aquire_lease, which then calls Lease.aquire_lease.

Currently all three of these functions are decorated - but only one of them makes a network call.

Reply:
I think this is the same problem as the `upload_blob` problem: #6492.